tests/test_scripts/mf6_water_balance/combine.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+
+from pathlib import Path
+from typing import Union
+
+import netCDF4 as nc
+import numpy as np
+import pandas as pd
+from test_scripts.mf6_water_balance.MF6_wbal_listing import listfile_to_dataframe
+
+
+def create_modflow_waterbalance_file(
+    mf_listfile: Path,
+    output_file_xlsx: Union[Path, None] = None,
+    output_file_netcdf: Union[Path, None] = None,
+    output_file_csv: Union[Path, None] = None,
+) -> None:
+    modflow_results_dataframe = listfile_to_dataframe(mf_listfile)
+
+    if output_file_netcdf is not None:
+        print("Writing NetCDF")
+        writeNC(output_file_netcdf, modflow_results_dataframe, singlevar=False)
+    if output_file_csv is not None:
+        print("Writing CSV")
+        writeCSV(output_file_csv, modflow_results_dataframe)
+    if output_file_xlsx is not None:
+        print("Writing XLSX")
+        writeXLS(output_file_xlsx, modflow_results_dataframe)
+
+
+def writeNC(ncname: Path, df: pd.DataFrame, singlevar: bool):
+    nvar = len(df.columns)
+    ds = nc.Dataset(ncname, "w")
```

Comment:
You can use a context manager as seen here: https://unidata.github.io/netcdf4-python/#reading-data-from-a-multi-file-netcdf-dataset

Reply:
done